src/attributes/classes.js

```diff
@@ -23,9 +39,7 @@ jQuery.fn.extend( {
 			} );
 		}
 
-		if ( typeof value === "string" && value ) {
-			classes = value.match( rnothtmlwhite ) || [];
```

Comment:
Arrays should just have this assignment skipped in favor of using the first parameter as `classes` and the rest of the code can then stay the same...